modules/nixos/environment.nix

```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}: let
+  inherit (lib.attrsets) mapAttrsToList;
+  inherit (lib.lists) isList;
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkOption;
+  inherit (lib.strings) concatStringsSep;
+  inherit (lib.types) attrsOf listOf oneOf int path str;
+
+  inherit (pkgs) writeShellScript;
+
+  cfg = config.environment;
+
+  writeEnvScript = let
+    toEnv = env:
+      if isList env
+      then concatStringsSep ":" (map toString env)
+      else toString env;
+  in
+    attrs:
+      writeShellScript "load-env"
+      (concatStringsSep "\n"
+        (mapAttrsToList (name: value: "export ${name}=\"${toEnv value}\"") attrs));
+in {
+  options.environment = {
+    sessionVariables = mkOption {
+      type = attrsOf (oneOf [(listOf (oneOf [int str path])) int str path]);
+      default = {};
+      example = {
+        EDITOR = "nvim";
+        VISUAL = "nvim";
+      };
+      description = ''
+        A set of environment variables used in the user environment.
+        These variables will be set as systemd user environment
+        variables, using `~/.profile`. The value of each
+        variable can be either a string or a list of strings. The
+        latter is concatenated, interspersed with colon
+        characters.
```

Comment:
Is describing the type really necessary? In generated documentation, however we generate it, the type field already does this. Though, you can still warn that a list of strings is concatenated etc.

Reply:
This has been copied from the NixOS module, but I'll make it more succinct. No need to make it redundant

Reply:
Also forgot to remove the systemd mention in the previous description. I've updated that as well in the newly pushed commit